src/main/java/Duke.java

```diff
@@ -1,10 +1,387 @@
+                String deadline = task.substring(split + 5);  //exception may occur
+                description = description.trim();
+                deadline = deadline.trim();
+                Event e = new Event(description, deadline);
+                this.todoList.add(e);
+                StringBuilder outputMessage = new StringBuilder("Task added:\n");
+                outputMessage.append("    " + e);
+                outputMessage.append("\n  There are " + this.todoList.size() + " tasks in the list.");
+                editor.write("E | 0 | " + description + " | " + deadline + "\n");
+                editor.flush();
+                Duke.print(outputMessage.toString());
+            } catch (IndexOutOfBoundsException e) { // happens when input is "event xx /at" with no time given
+                Duke.print("Error: no event time provided");
+            } catch (IOException e){
+                Duke.print("File write error");
+            }
+        }
+    }
+
+    private void addDeadline(String task) {
+        int split = task.indexOf(" /by");
+        if (split == -1) {
+            Duke.print("Error: deadline not given. Specify deadline using \"/by\"");
+        } else {
+            try {
+                String description = task.substring(0, split);
+                String deadline = task.substring(split + 5);
+                description = description.trim();
+                deadline = deadline.trim();
+                Deadline d = new Deadline(description, deadline);
+                this.todoList.add(d);
+                StringBuilder outputMessage = new StringBuilder("Task added:\n");
+                outputMessage.append("    " + d);
+                outputMessage.append("\n  There are " + this.todoList.size() + " tasks in the list.");
+                editor.write("D | 0 | " + description + " | " + deadline + "\n");
+                editor.flush();
+                Duke.print(outputMessage.toString());
+            } catch (IndexOutOfBoundsException e) { //same as event time
+                Duke.print("Error: no deadline provided");
+            } catch (IOException e){
+                Duke.print("File write error");
+            }
+        }
+    }
+
+    private void raiseSingleWordInput(String badInput) throws DukeException {
+        if (badInput.equals("todo") || badInput.equals("event") || badInput.equals("deadline")) {
+            throw new DukeException("Error: no description for task.");
+        } else if (badInput.equals("done")) {
+            throw new DukeException("Error: done task index missing");
+        } else {
+            throw new DukeException("Error: no such command");
+        }
+    }
+
+    public class Task {
+        protected String description;
+        protected boolean isDone;
+
+        public Task(String description) {
+            this.description = description;
+            this.isDone = false;
+        }
+
+        public void setDone() {
```

Comment:
Consider decoupling the printing of the message. The task should focus on being a task in order to be extensible.

Reply:
Also, what if I want to unmark the task as complete?